File: android-bluetooth/src/main/java/com/lightningkite/kwift/bluetooth/BleServer.shared.kt
package com.lightningkite.kwift.bluetooth

import com.lightningkite.kwift.bytes.Data
import com.lightningkite.kwift.observables.*
import io.reactivex.disposables.Disposable
import java.util.*

//My implementation

interface BleServer: Disposable {
    val characteristics: Map<UUID, Map<UUID, BleCharacteristicServer>>
    var advertising: Boolean
}

interface BleClient {
    val info: BleDeviceInfo
    val connected: Boolean
    fun respond(request: RequestId, data: Data, status: BleResponseStatus)
    /** Does not require a confirmation from the device. **/
    fun notify(service: UUID, characteristic: UUID, value: Data)
    /** Requires a confirmation from the device. **/
    fun indicate(service: UUID, characteristic: UUID, value: Data)
}

File: ios-bluetooth/KwiftBluetooth/com/lightningkite/kwift/bluetooth/BleServer.shared.swift
//Package: com.lightningkite.kwift.bluetooth
//Converted using Kwift2

import Foundation
import RxSwift
import RxRelay
import Kwift



public protocol BleServer: Disposable {
    
    var characteristics: Dictionary<UUID, Dictionary<UUID, BleCharacteristicServer>> { get }
    
    var advertising: Bool { get set }
}
 
 

public protocol BleClient {
    
    var info: BleDeviceInfo { get }
    
    var connected: Bool { get }
    
    func respond(request: RequestId, data: Data, status: BleResponseStatus) -> Void
    func respond(_ request: RequestId, _ data: Data, _ status: BleResponseStatus) -> Void
    
    func notify(service: UUID, characteristic: UUID, value: Data) -> Void
    func notify(_ service: UUID, _ characteristic: UUID, _ value: Data) -> Void
    
    func indicate(service: UUID, characteristic: UUID, value: Data) -> Void
    func indicate(_ service: UUID, _ characteristic: UUID, _ value: Data) -> Void
}
 
